packages/perseus/src/widgets/interactive-graphs/math/box.ts

```diff
@@ -0,0 +1,41 @@
+import {clamp} from "./clamp";
+import {X, Y} from "./coordinates";
+import {MAX, MIN} from "./interval";
+
+import type {Interval, vec} from "mafs";
+
+export type Box = [x: Interval, y: Interval];
+
+// Restricts the `point` to be within `box`, by clamping each coordinate to the
+// corresponding interval.
+export function clampToBox(box: Box, point: vec.Vector2): vec.Vector2 {
+    return [clamp(point[X], ...box[X]), clamp(point[Y], ...box[Y])];
+}
+
+// Reduces the size of `box`, trimming each corner by the given positive
+// `amount`. If a component of `amount` is negative, the box is expanded in
+// that dimension instead.
+export function inset(amount: vec.Vector2, box: Box): Box {
+    return ensureValid([
+        [box[X][MIN] + amount[X], box[X][MAX] - amount[X]],
+        [box[Y][MIN] + amount[Y], box[Y][MAX] - amount[Y]],
+    ]);
+}
+
+export function ensureValid(box: Box): Box {
```

Comment:
I like the comments for the other functions. Perhaps we could have one here, too? Mostly because I don't know what setting the min/max values to the average does.

Reply:
I think `ensureValid` is the wrong abstraction. The behavior of setting the values to the average is specific to `inset`. The purpose of setting min and max to the average is that the center of the box should not be changed by `inset`.

Reply:
I've reworked this code. Let me know what you think!

Reply:
That makes a LOT more sense! Thank you!!